.github/workflows/upgrade-testing-run.yml

```diff
@@ -0,0 +1,94 @@
+##
+# A callable workflow that runs WordPress upgrade testing under the conditions provided.
+##
+name: Upgrade Tests
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        description: 'Operating system to run tests on'
+        required: false
+        type: 'string'
+        default: 'ubuntu-latest'
+      wp:
+        description: 'The version of WordPress to start with.'
+        required: true
+        type: 'string'
+      new-version:
+        description: 'The version of WordPress to update to.'
+        type: 'string'
+      php:
+        description: 'The version of PHP to use, in the format of X.Y'
+        required: true
+        type: 'string'
+      multisite:
+        description: 'Whether to run tests as multisite'
+        required: false
+        type: 'boolean'
+        default: false
+      db-type:
+        description: 'Database type. Valid types are mysql and mariadb'
+        required: false
+        type: 'string'
+        default: 'mysql'
+      db-version:
+        description: 'Database version'
+        required: false
+        type: 'string'
+        default: '5.7'
+
+jobs:
+  # Runs upgrade tests on a build of WordPress.
+  #
+  # Performs the following steps:
+  # - Checks out the repository.
+  # - Sets up PHP.
+  # - Starts the database server.
+  # - Downloads the specified version of WordPress.
+  # - Creates a `wp-config.php` file.
+  # - Installs WordPress.
+  # - Updates to the latest minor version.
+  # - Updates to the version of WordPres being tested.
+  upgrade-tests:
+    name: ${{ inputs.wp }} to ${{ inputs.new-version && inputs.new-version || 'latest' }} / PHP ${{ inputs.php }} with ${{ 'mariadb' == inputs.db-type && 'MariaDB' || 'MySQL' }} ${{ inputs.db-version }}${{ inputs.multisite && ' multisite' || '' }}
+    runs-on: ${{ inputs.os }}
+    timeout-minutes: 20
+
+    services:
+      database:
```

Comment:
I changed the service name to `database` to make it easier to add other database types later. We should be able to add MariaDB testing fairly easily.